store/copr/mpp.go

```diff
@@ -123,15 +123,31 @@ type mppIterator struct {
 
 	respChan chan *mppResponse
 
-	rpcCancel *tikv.RPCCanceller
+	cancelFunc context.CancelFunc
 
 	wg sync.WaitGroup
 
 	closed uint32
+
+	vars *kv.Variables
+
+	mu sync.Mutex
 }
 
 func (m *mppIterator) run(ctx context.Context) {
 	for _, task := range m.tasks {
+		if atomic.LoadUint32(&m.closed) == 1 {
+			break
+		}
+		m.mu.Lock()
+		switch task.State {
```

Comment:
state should be atomic type so that we don't have to use mutex

Reply:
the mutex has another work: ensure the `cancelMppTasks` juste being called exclusively. besides, there are not atomic enum type.